src/services/role/format.ts

```diff
@@ -0,0 +1,39 @@
+import cuid from 'cuid'
+import { V1Role } from '@kubernetes/client-node'
+import { K8sRole } from '../../types/generated'
+import formatMetadata from '../../util/metadata'
+
+export default ({
+  entity,
+  context,
+}: {
+  entity: V1Role
+  context: string
+}): K8sRole => {
+  const {
+    apiVersion,
+    kind,
+    metadata,
+    rules
+  } = entity
+
+  const formattedMetadata = formatMetadata(metadata)
+
+  const mappedRules = rules?.map(({ apiGroups, nonResourceURLs, resources, resourceNames, verbs }) => ({
+    id: cuid(),
+    apiGroups,
```

Comment:
I guess `apiGroups` comes across like this but its a bit weird to see in the output

```
            "apiGroups": [
              ""
            ],
```

Reply:
yea Idk if there is much I can do there. idk why it is setup that way. I suppose I could filter out empty strings from the array but maybe there is a reason for it.